.github/workflows/java.yml

```diff
@@ -25,48 +25,46 @@ on:
             - .github/workflows/install-valkey/action.yml
             - .github/json_matrices/build-matrix.json
     workflow_dispatch:
+        inputs:
+            full-matrix:
+                description: "Run the full matrix"
+                required: false
+                default: "false"
+
+    schedule:
+        - cron: "0 0 * * *"
 
 concurrency:
     group: java-${{ github.head_ref || github.ref }}
     cancel-in-progress: true
 
 jobs:
-    load-engine-matrix:
+    get-matrices:
         runs-on: ubuntu-latest
+        # Avoid running on schedule for forks
+        if: (github.repository_owner == 'valkey-io' || github.event_name != 'schedule')
```

Comment:
I suspect the condition is incorrect. Keep in mind that job should still run on fork on PR/push/trigger.

Reply:
Same in other workflows